src/github.com/couchbase/sync_gateway/rest/changes_api_test.go

```diff
@@ -519,6 +520,65 @@ func TestDocDeduplication(t *testing.T) {
 
 }
 
+func TestIndexChangesMultipleRevisions(t *testing.T) {
+	it := initIndexTester(true, `function(doc) {channel(doc.channel);}`)
```

Comment:
This was my attempt to reproduce the intermittent functional test failure, but wasn't able to reproduce even when running with -race.  Leaving the test here as it's a useful scenario that's not otherwise covered.